tests/test-cases.js

```diff
@@ -1458,6 +1458,49 @@ var CUSTOM_TESTS = [
 				}
 			});
 		}
+	},
+	{
+		description: "Should not assume we own the worker unless papaworker is in the search string",
+		expected: [false, true, true, true, true],
+		run: function(callback) {
+			if (typeof Worker === 'undefined') {
+				// We can't test workers in this environment, so pass.
+				callback([false, true, true, true, true]);
+				return;
+			}
+			var searchStrings = [
+				'',
+				'?papaworker',
+				'?x=1&papaworker',
+				'?x=1&papaworker&y=1',
+				'?x=1&papaworker=1'
+			];
+			var results = searchStrings.map(function () { return false; });
+
+			var numResults = 0;
+			var setResult = function (idx, result) {
+				results[idx] = result;
+				numResults += 1;
+				if (numResults === results.length) {
+					callback(results);
+				}
+			};
+
+			searchStrings.forEach(function (searchString, idx) {
+				var w = new Worker('../papaparse.js' + searchString);
+				// Give it .5s to do something
+				var timeout = setTimeout(function () {
+					w.terminate();
+					setResult(idx, false);
+				}, 500);
```

Comment:
You could simplify this a bit by initializing to false and having a single timeout with callback(results), if you like.